### PR TITLE
bpo-30978: str.format_map() now passes key lookup exceptions through.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**

### CODE REVIEW

**Comment by vstinner** [comment]
A test fails:
======================================================================

ERROR: test_match_getitem (test.test_re.ReTests)

----------------------------------------------------------------------

Traceback (most recent call last):

  File "/home/travis/build/python/cpython/Lib/test/test_re.py", line 472, in test_match_getitem

    'a1={a2}'.format_map(m)

IndexError: no such group


You should document this behaviour change in the NEWS entry (and fix the test).

**Comment by ericvsmith** [comment]
Other than the failing test in test_re, this looks good to me.

**Review by vstinner — APPROVED** [review]
Thanks, it's now better with NEWS describing the old behaviour ;-)

File: Misc/NEWS.d/next/Core and Builtins/2017-07-21-07-39-05.bpo-30978.f0jODc.rst [hunk 1]
```diff
@@ -1 +1,2 @@
 str.format_map() now passes key lookup exceptions through.
+Previousle any exception was replaced with a KeyError exception.
```
Comment (ericvsmith) [inline Misc/NEWS.d/next/Core and Builtins/2017-07-21-07-39-05.bpo-30978.f0jODc.rst]: Typo: Previousle -> Previously

Comment (serhiy-storchaka) [reply Misc/NEWS.d/next/Core and Builtins/2017-07-21-07-39-05.bpo-30978.f0jODc.rst]: Thanks!

**Review by rhettinger — APPROVED** [review]

**Pull request merged**